common/src/View/MapFrame.cpp

```diff
@@ -1193,7 +1193,21 @@ namespace TrenchBroom {
             if (IsBeingDeleted()) return;
 
             if (canDoCsgConvexMerge()) { // on gtk, menu shortcuts remain enabled even if the menu item is disabled
-                m_document->csgConvexMerge();
+                if (m_mapView->vertexToolActive()) {
+                    m_document->csgConvexMerge(m_mapView->vertexTool()->handleManager().selectedHandles());
+                } else if (m_mapView->edgeToolActive()) {
+                    std::vector<vm::vec3> vertices;
```

Comment:
Instead of exposing the map document to the handle manager and changes thereof, U would suggest to delegate the merge to these tools.

Reply:
I've done this in my latest commit.